src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,201 @@
+            toRemoveNode.prev.next = insteadNode;
+            toRemoveNode.next.prev = insteadNode;
+            return toRemoveNode.body;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexForBound(index);
+        Node<T> toRemoveNode = getNode(index);
+        if (index == 0) {
+            head = getNode(1);
+        } else if (index == size - 1) {
+            tail = getNode(size - 2);
+        }
+        unlinkNode(toRemoveNode);
+        size--;
+        return toRemoveNode.body;
     }
 
     @Override
     public boolean remove(T object) {
+        for (int i = 0; i < size; i++) {
+            if (get(i) == object || get(i) != null && get(i).equals(object)) {
+                remove(i);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void linkNewNode(Node<T> tempNode, T value) {
+        Node<T> toAdd = new Node<>(tempNode.prev, value, tempNode);
+        tempNode.prev.next = toAdd;
+        tempNode.prev = toAdd;
+        size++;
+    }
+
+    private void checkIndexForBound(int index) {
+        String exception = String.format("Wrong index! %d index is out of bounds.", index);
+        if (index < 0 || index > size - 1 && size > 1) {
+            throw new IndexOutOfBoundsException(exception);
+        }
+    }
+
+    private void checkIndexForBoundAdd(int index) {
+        String exception = String.format("Wrong index! %d index is out of bounds.", index);
+        if (index < 0 || index > size && size > 1) {
+            throw new IndexOutOfBoundsException(exception);
+        }
+    }
+
+    private void unlinkNode(Node<T> node) {
+        Node<T> nextNode = node.next;
+        Node<T> prevNode = node.prev;
+        if (prevNode != null) {
+            prevNode.next = nextNode;
+        }
+        if (nextNode != null) {
+            nextNode.prev = prevNode;
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        if (index == 0) {
+            return head;
+        }
+        if (index < (size - 1) / 2) {
+            int indexCounter = 0;
+            Node<T> tempNode = head.next;
+            while (indexCounter != index) {
+                indexCounter++;
+                if (indexCounter == index) {
+                    return tempNode;
+                }
+                tempNode = tempNode.next;
+
+            }
+        } else {
+            int indexCounter = size;
+            Node<T> tempNode = tail;
+            while (indexCounter != index) {
+                indexCounter--;
+                if (indexCounter == index) {
+                    return tempNode;
+                }
+                tempNode = tempNode.prev;
+            }
+        }
+        return null;
```

Comment:
this is too verbose, this method can be done in like 15 lines. Use regular for inside of for-each clauses and only one return at the end, call checker method here,so that we know it won't return null - for a valid index there should always be some node